domene/src/main/java/no/nav/foreldrepenger/selvbetjening/innsending/VedleggReferanseMapperTjeneste.java

```diff
@@ -0,0 +1,128 @@
+                case UTTAK -> throw new UnsupportedOperationException("Utviklerfeil: Svangerskapspenger har ikke uttak!");
+                case TILRETTELEGGING -> leggVedleggTilTilrettelegging(foreldrepenger.tilrettelegging(), vedlegg);
+            }
+        }
+    }
+
+
+
+    private static void leggVedleggsreferanserTilSøknad(no.nav.foreldrepenger.selvbetjening.kontrakt.innsending.v2.dto.engangsstønad.EngangsstønadDto foreldrepenger) {
+        for (var vedlegg : foreldrepenger.vedlegg()) {
+            switch (vedlegg.getDokumenterer().type()) {
+                case BARN -> leggVedleggTilBarn(foreldrepenger.barn(), vedlegg);
+                case OPPTJENING -> throw new UnsupportedOperationException("Utviklerfeil: Engangstønad har ikke annen inntektskilde!");
+                case UTTAK -> throw new UnsupportedOperationException("Utviklerfeil: Engangstønad har ikke uttak!");
+                case TILRETTELEGGING -> throw new UnsupportedOperationException("Utviklerfeil: Engangstønad har ikke tilrettelegging!");
+            }
+        }
+    }
+
+    private static void leggVedleggsreferanserTilSøknad(EngangsstønadDto foreldrepenger) {
+        for (var vedlegg : foreldrepenger.vedlegg()) {
+            switch (vedlegg.getDokumenterer().type()) {
+                case BARN -> leggVedleggTilBarn(foreldrepenger.barn(), vedlegg);
+                case OPPTJENING -> leggVedleggTilAnnenInntekt(foreldrepenger.søker().andreInntekterSiste10Mnd(), vedlegg);
+                case UTTAK -> throw new UnsupportedOperationException("Utviklerfeil: Engangstønad har ikke uttak!");
+                case TILRETTELEGGING -> throw new UnsupportedOperationException("Utviklerfeil: Engangstønad har ikke tilrettelegging!");
+            }
+        }
+    }
+
+    private static void leggVedleggsreferanserTilSøknad(ForeldrepengesøknadDto foreldrepenger) {
+        for (var vedlegg : foreldrepenger.vedlegg()) {
+            switch (vedlegg.getDokumenterer().type()) {
+                case BARN -> leggVedleggTilBarn(foreldrepenger.barn(), vedlegg);
+                case OPPTJENING -> leggVedleggTilAnnenInntekt(foreldrepenger.søker().andreInntekterSiste10Mnd(), vedlegg);
+                case UTTAK -> leggVedleggTilUttak(foreldrepenger.uttaksplan(), vedlegg);
+                case TILRETTELEGGING -> throw new UnsupportedOperationException("Utviklerfeil: Foreldrepenger har ikke tilrettelegging!");
+            }
+        }
+    }
+
+    private static void leggVedleggTilTilrettelegging(List<TilretteleggingDto> tilrettelegging, VedleggDto vedlegg) {
+        var gjeldendeTilrettelegging = tilrettelegging.stream()
+            .filter(t -> t.arbeidsforhold().equals(vedlegg.getDokumenterer().arbeidsforhold()))
+            .findFirst()
+            .orElseThrow(() -> new IllegalStateException("Utviklerfeil: klarte ikke finne arbeidsforhold i tilrettelegginger"));
```

Comment:
Her bør du se på SVP for næringsdrivende. Kommer de med orgnummer i arbeidsforhold?
Hvilke kategorier kan man laste opp dok for i søknad - arbeid, næring, begge?